### PR TITLE
py/objslice: make slice attributes readable

### PR DESCRIPTION
**Pull request by pramasoul**
Provides access to the `.start`, `.stop`, and `.step` attributes of slice objects. This is needed for classes that implement `__getitem__()`, to respond intelligently to a slice access.

From https://docs.python.org/3/reference/datamodel.html:

> Slice objects are used to represent slices for `__getitem__()` methods. They are also created by the built-in `slice()` function.
> 
> Special read-only attributes: start is the lower bound; stop is the upper bound; step is the step value; each is `None` if omitted.

The cost is 32 bytes of text segment in the stmhal build. No RAM used.
I've configured it out in `py/mpconfig.h`, and then enabled in for unix and stmhal ports. Should it be this way, or should it be in by default?

### CODE REVIEW

**Comment by pfalcon** [comment]
> Should it be this way, or should it be in by default?

I'd say this way is ok.

File: py/objslice.c [hunk 1]
```diff
@@ -57,10 +57,30 @@ STATIC void slice_print(const mp_print_t *print, mp_obj_t o_in, mp_print_kind_t
...
 
+#if MICROPY_PY_BUILTINS_SLICE_ATTRS
+STATIC void slice_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
+    if (dest[0] != MP_OBJ_NULL) {
+        // not load attribute
+        return;
+    }
+    mp_obj_slice_t *self = self_in;
+    if (attr == MP_QSTR_start) {
+	    dest[0] = self->start;
```
Comment (dpgeorge) [inline py/objslice.c]: Indent level.

File: tests/basics/slice_attrs.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# print slice attributes
+class A:
+    def __getitem__(self, idx):
+        print(idx.start, idx.stop, idx.step)
+
+# The skip test method for qemu-arm/test-main.c requires we test
+# for the feature being testable and bail thusly. It does not use
+# the tests/feature_check mechanism used by tests/run-tests
+try:
+    t = a()[1:2]
```
Comment (dpgeorge) [inline tests/basics/slice_attrs.py]: Should be capital A.

File: tests/feature_check/slice_attrs_check.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class A:
```
Comment (dpgeorge) [inline tests/feature_check/slice_attrs_check.py]: Feature checks are for things that can't be simply skipped by printing SKIP in the file.  So in this case don't have a feature check script and just use the SKIP mechanism as you've done above.

**Comment by pfalcon** [comment]
Ok, please squash out *irrelevant* commits (e.g. README change should stay its own commit, and really doesn't belong to this PR).

**Comment by pramasoul** [comment]
Done

**Comment by dpgeorge** [comment]
Merged in aeb62f9ae3169d789800591eea390e020c7532bf.

**Pull request closed without merging**